platform-hub-api/app/controllers/kubernetes/tokens_controller.rb

```diff
@@ -0,0 +1,75 @@
+class Kubernetes::TokensController < ApiJsonController
+
+  before_action :find_identity
+
+  # GET /:user_id
+  def index
+    authorize! :read, @identity
+    render json: Kubernetes::TokenService.tokens_from_identity(@identity)
+  end
+
+  # PATCH/PUT /:user_id/:cluster
+  def create_or_update
+    authorize! :manage, @identity
+    tokens, created_or_updated_token = Kubernetes::TokenService.create_or_update_token(
+      @identity,
+      params[:cluster],
+      params[:token][:groups]
```

Comment:
Yeah, fair enough. Just thinking in terms of a standalone API, and guarding from bad API requests (either through bugs or maliciousness).